Source/WebKit/Shared/API/APIFrameHandle.h

```diff
@@ -29,26 +29,32 @@
 #include <WebCore/FrameIdentifier.h>
 #include <wtf/Ref.h>
 
-namespace IPC {
-class Decoder;
-class Encoder;
-}
-
 namespace API {
 
 class FrameHandle final : public ObjectImpl<Object::Type::FrameHandle> {
 public:
-    static Ref<FrameHandle> create(WebCore::FrameIdentifier);
-    static Ref<FrameHandle> createAutoconverting(WebCore::FrameIdentifier);
-
-    explicit FrameHandle(WebCore::FrameIdentifier, bool isAutoconverting);
+    static Ref<FrameHandle> create(WebCore::FrameIdentifier frameID)
+    {
+        return adoptRef(*new FrameHandle(frameID, false));
+    }
+    static Ref<FrameHandle> createAutoconverting(WebCore::FrameIdentifier frameID)
+    {
+        return adoptRef(*new FrameHandle(frameID, true));
+    }
+    static Ref<FrameHandle> create(WebCore::FrameIdentifier frameID, bool autoconverting)
+    {
+        return adoptRef(*new FrameHandle(frameID, autoconverting));
+    }
+
+    explicit FrameHandle(WebCore::FrameIdentifier frameID, bool isAutoconverting)
```

Comment:
This constructor should be private.

Reply:
That doesn't work because APIObject's `constructInWrapper` expects access to the constructor. I don't understand the memory handling of that function, so don't easily see how to change it to use a `create` method.